crates/walrus-service/src/node/storage/blob_info.rs

```diff
@@ -1683,8 +1683,21 @@ impl Mergeable for BlobInfoV1 {
                 epoch,
                 event: status_event,
             }),
+            BlobInfoMergeOperand::MarkMetadataStored(is_metadata_stored) => {
+                tracing::info!(
+                    is_metadata_stored,
+                    "marking metadata stored for an untracked blob ID; blob info will be removed \
```

Comment:
question: why blob info will be removed? Isn't that the blob info has already be gone, which triggers the issue?

Reply:
The blob info had been deleted previously, but with this merge operand it will be recreated. During GC, we notice that this refers to a blob with no non-expired objects, so we will remove both the blob info and the related metadata.